### ISSUE TITLE
`UI.refresh_inspectors` does not refresh Entity Info on Windows (Qt)

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#880]
#### SketchUp Ruby API Issue

**SketchUp Version**: `2023.0.0` and `2023.0.1`
**Platform**: Windows using Qt framework

[`UI.refresh_inspectors`](https://ruby.sketchup.com/UI.html#refresh_inspectors-class_method) does **not** refresh Entity Info inspector.

Entity Info's material swatch fails to refresh for a very common class of direct selections.

It may have also been reported that the tag (layer) picklist control is not being updated.
The other controls on the Entity Info inspector should also be checked (the toggle buttons and the advanced attribute values.)

#### Testing:

[entity_info_test.skp.zip](https://github.com/SketchUp/api-issue-tracker/files/10986490/entity_info_test.skp.zip)

**Steps to Reproduce**:
1. Create 2 different entities of the same kind, e.g. faces, edges, surfaces, groups, components – any geometry capable of displaying material.
2. Paint the 2 different entities with 2 different materials.
3. Select one of the painted entities.
4. Directly select the other and differently painted entity.

**Note that Entity Info now fails to update its material swatch.**
For the current selection, Entity Info now is displaying an incorrect material.

**Notice that with the Ruby Console or Model Info open, as soon as you click the window and give it focus, the Entity Info inspector refreshes.**

Now copy the following into the console ...
```ruby
  class SelectionSpy
    def onSelectionBulkChange(selection)
      puts "Selection change"
      UI.refresh_inspectors
    end
  end
  Sketchup.active_model.selection,add_observer(SelectionSpy.new)
```
... and repeat the test.

**Notice that the method call does not work.**

Neither does using `Sketchup.focus`.

----

Weirdly, what **DOES** work is to switch the focus to one of SketchUp's floating child windows, such as Ruby Console or Model Info window.

I wrote up a quick little observer plugin that creates a hidden HTML dialog for a half second then closes. This causes focus to shift between the dialog and then back to SketchUp which triggers the Entity Info refresh.

The delay is needed otherwise the Entity Info inspector becomes confused about what is selected.

[EntityInfo_refresh.zip](https://github.com/SketchUp/api-issue-tracker/files/10986366/EntityInfo_refresh.zip)

But the following more simplified observer using SketchUp's console window **also works**.
Again, it seems that a delay is needed for SketchUp to "catch up" will some internal focusing issue.

```ruby
  class SelectionSpy
    def onSelectionBulkChange(selection)
      puts "Selection change"
      SKETCHUP_CONSOLE.hide if SKETCHUP_CONSOLE.visible?
      sleep(0.5)
      SKETCHUP_CONSOLE.show
    end
  end
  Sketchup.active_model.selection.add_observer(SelectionSpy.new)
```
~

**Comment by Fredosixx** [comment]
I think this was noticed and reported during the Beta tests. But now, it's good to have it in the issue database.

I confirm this is still there event with SU2023.0.1

**Comment by DanRathbun** [comment]
I was not sure, but since it hasn't been fixed in initial release or the first update, it needs to be a **publicly** known issue.

**Comment by thomthom** [comment]
> I think this was noticed and reported during the Beta tests.

Can you send me an email/message with link to this thread/report?

**Comment by Fredosixx** [comment]
Forget the beta.

The problem is still there and well described by Dan. This should be enough.

**Comment by DanRathbun** [comment]
> Can you send me an email/message with link to this thread/report?

I sent a forum PM. The 2 reports are recent and may be post 23.0.1.
I cut and pasted some of the 1st report into this issue's initial post on reproducing the issue.

**Comment by thomthom** [comment]
> Forget the beta.
> The problem is still there and well described by Dan. This should be enough.

I'm trying to track down the communication chain, how and why we might have missed the message.

> I sent a forum PM. The 2 reports are recent and may be post 23.0.1.

To my `tt_su` forum account?

**Comment by DanRathbun** [comment]
> To my `tt_su` forum account?

Sorry, yes but I forgot to click the send button. It's sent now.

**Comment by thomthom** [comment]
I don't think this is an API bug, since it fails to update on a normal selection - as it used to do in SU2022 and earlier. But I'm bumping this internally.

**Comment by DanRathbun** [comment]
Okay, thank you!

The impetus for this issue was an attempt to work around the GUI bug that was not firing a refresh of inspector dialogs.
I was hoping that the Ruby method made a direct call to the C-side routine that does the refreshing.

But, if it is the refreshing routine(s) themselves that are bugged, this would not work.

**Comment by thomthom** [comment]
Yes, that's what it looked like. Btw, on the latest develop branch internally this appear to be working correctly.
